### PR TITLE
added cpp code for shell sort

### PR DESCRIPTION
**Pull request by anshmishra010**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

added cpp code for shell sort

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#304

### CODE REVIEW

**Comment by anshmishra010** [comment]
@siddharth25pandey  please see this

File: docs/sorting-algorithms/shell-sort.mdx [hunk 1]
```diff
@@ -132,6 +133,58 @@ int main()
...
+    }
+    shellSort(sort,size);
+    cout<<"Sorted Array in ascending order is :";
+    for(int i=0;i<size;i++){
+        cout<<sort[i]<<" ";
+    }
+    cout<<endl;
+    return 0;
+}
+
```
Comment (siddharth25pandey) [inline docs/sorting-algorithms/shell-sort.mdx:184]: Add the Time  & Space Complexity for the Shell Sort

File: docs/sorting-algorithms/shell-sort.mdx [hunk 2]
```diff
@@ -132,6 +133,58 @@ int main()
...
+This algorithm avoids large shifts as in case of insertion sort, if the smaller value is to the far right 
+and has to be moved to the far left.This algorithm uses insertion sort on a widely spread elements, first to sort 
+them and then sorts the less widely spaced elements. This spacing is termed as interval.
+Worst case time complexity = O(n2)
+Best case complexity = O(nlog(n)).
+Space complexity = O(1).
+*/
+
+
+#include<iostream>
```
Comment (siddharth25pandey) [inline docs/sorting-algorithms/shell-sort.mdx:151]: Use **bits/stdc**

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@anshmishra010 Small changes required

**Comment by anshmishra010** [comment]
@siddharth25pandey  done

**Review by siddharth25pandey — APPROVED** [review]
@anshmishra010 Perfect one !

**Pull request merged**